sdk/src/trace/tracer.cc

```diff
@@ -8,6 +8,16 @@ namespace sdk
 {
 namespace trace
 {
+void Tracer::SetProcessor(std::shared_ptr<SpanProcessor> processor) noexcept
+{
+  processor_ = processor;
```

Comment:
I think you want something like `std::atomic_store(&processor_, processor)`

See https://en.cppreference.com/w/cpp/memory/shared_ptr/atomic